### DISCUSSION TITLE
q-select q-select shows values ​​instead of labels after adding

### DISCUSSION THREAD

**Discussion by oneeoote** [quasarframework/quasar#12865] [General - Components / Directives / etc]
### What happened?

When v-model is an object, use use-input and new-value, and q-select the label value to display instead of label after adding

### What did you expect to happen?

You want q-SELECT to display labels after you add them

### Reproduction URL

https://codepen.io/oneeoote/pen/vYpKGqW

### How to reproduce?

```
<div id="q-app">
  <div>
    <q-select v-model="model" outlined use-input :options="options" option-value="id" option-label="name" emit-value map-options label="UNIT" input-debounce="300" @new-value="addUnit" @filter="filterUnits">
      <template v-slot:no-option>
        <q-item>
          <q-item-section class="text-grey">Press Enter to add</q-item-section>
        </q-item>
      </template>
    </q-select>
  </div>
  <p>model value: {{model}}</p>
</div>
```

```
const stringOptions = [
  { id: 1, name: "11111" },
  { id: 2, name: "222222" }
];
new Vue({
  el: "#q-app",
  data: {
    model: '',
    options: stringOptions,
    id: 3
  },
  methods: {
    filterUnits(val, update) {
      if (val === "") {
        update(() => {
          this.options = stringOptions;
        });
        return;
      }

      update(() => {
        const needle = val.toLowerCase();
        this.options = stringOptions.filter(
          (v) => v.name.toLowerCase().indexOf(needle) > -1
        );
      });
    },
    addUnit(val, done) {
      done({ id: this.id, name: val });
      this.id++;
    }
  }
});
```

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome

**Comment by pdanpdan** [comment]
you don't add the new option in the list of options (neither in stringOptions nor in this.options)

https://codepen.io/pdanpdan/pen/bGaeeQG?editors=1010

**Comment by oneeoote** [comment]
Yes, thank you very much, my development environment is much more complicated than the example. According to your answer, it should take a long time after adding it. I will try adding a setTimeout to the done method.
thanks again

**Comment by oneeoote** [comment]
There is another question, my initial value is a value that does not exist in options, can it not be displayed in q-select? Do not display the value that does not exist in options

demo:
[https://codepen.io/oneeoote/pen/vYpKGqW](https://codepen.io/oneeoote/pen/vYpKGqW)
